lib/slither/generator.rb

```diff
@@ -1,7 +1,9 @@
 class Slither
   class Generator
-    def initialize(definition)
+    def initialize(definition, trailing_newline = false)
       @definition = definition
+      @generator_should_add_trailing_newline = trailing_newline
+      @force_generator_to_add_newline = definition.options[:trailing_newline]
```

Comment:
Can you explain why there are 2 ways to configure trailing new line? Is it not sufficient to have it just in the options?

That means can we just have one place to configure it:

```ruby
Slither.define :test, trailing_newline: true do
end
```

Reply:
As i have seen we have a wrapper class for slither in the applications. I wanted to provide a generator level settings which would effect every slither method.
Also I wanted to provide a setting on a definition to help the change. With this approach we would be able to change it one by on to have a newline  for each file type and then just toggle the generator level when all of them has the trailing newline.
Did i overthought it? 🙃

Reply:
Ah okay. I think you explained it well and it's a good idea. Maybe you could include some comments on how to use this either in the README or as in-line comment?